plugin/storage/badger/factory.go

```diff
@@ -111,6 +111,13 @@ func (f *Factory) Initialize(metricsFactory metrics.Factory, logger *zap.Logger)
 	}
 	f.store = store
 
+	err = badgerStore.SchemaUpdate(store, logger)
```

Comment:
Is it possible to run the badger with collector (rw) in one host and query (read-only) in another? If so, you'd need a way to lock the storage here, so that only one pod will attempt to update the schema.

Reply:
Either lock, or a CLI option to to skip the schema migrations.

Reply:
Badger creates a lock when it reads the storage files, thus no two instances of badger can share the files. Also, the updates happen in a transaction -> they either completely succeed or fail (ACID guarantees) - thus preventing any issues with concurrent access in the same process.

So there's no additional locking required. And no, you can't access same storage files from two different hosts as they can't share the memory (which has the current state before it is written to the disk). While I guess WAL polling could be done, the LSM tree's compactions would make it next to impossible. I'm not really aware of any real storage engine that would allow it.

Reply:
Thanks for the clarification.